src/Logging.XUnit/XUnitLogger.cs

```diff
@@ -213,7 +182,20 @@ public virtual void WriteMessage(LogLevel logLevel, int eventId, string? message
 
             try
             {
-                outputHelper.WriteLine($"[{Clock():u}] {logLevelString}{formatted}");
+                ITestOutputHelper? outputHelper = _outputHelperAccessor?.OutputHelper;
+                IMessageSink? messageSink = _messageSinkAccessor?.MessageSink;
+
+                var line = $"[{Clock():u}] {logLevelString}{formatted}";
+                if (outputHelper != null)
+                {
+                    outputHelper.WriteLine(line);
+                }
+
+                if (messageSink != null)
+                {
+                    var sinkMessage = MessageSinkMessageFactory(line);
```

Comment:
Why not just access the `_messageSinkMessageFactory` field directly?

Reply:
Indeed, addressed in 4361d4549b5e0224b281d690aac9ee754656f390.